src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -422,23 +559,21 @@ private static double computeTigExplainQualOfOneConfiguration(final List<Alignme
         final List<GoodAndBadMappings> bestConfigurations = nameSeqAndBestConfigurationsOfOneAssemblyContig._2;
         if (bestConfigurations.size() > 1) { // more than one best configuration
             return bestConfigurations.stream()
-                    .map(configuration -> updateContigMappingsWithGapSplit(contigName, contigSeq, configuration, true))
+                    .map(mappings -> splitGaps(mappings, false))
+                    .map(mappings -> removeNonUniqueMappings(mappings, ALIGNMENT_MQ_THRESHOLD, ALIGNMENT_LOW_READ_UNIQUENESS_THRESHOLD))
+                    .map(mappings -> makeContig(contigName, contigSeq, mappings, true))
                     .sorted(getConfigurationComparator())
                     .iterator();
         } else {
-            return Collections.singletonList(updateContigMappingsWithGapSplit(contigName, contigSeq, bestConfigurations.get(0), false)).iterator();
+            final GoodAndBadMappings intermediate = splitGaps(bestConfigurations.get(0), false);
+            final GoodAndBadMappings result = removeNonUniqueMappings(intermediate, ALIGNMENT_MQ_THRESHOLD, ALIGNMENT_LOW_READ_UNIQUENESS_THRESHOLD);
+            return Collections.singletonList(makeContig(contigName, contigSeq, result, false)).iterator();
         }
     }
 
-    private static AssemblyContigWithFineTunedAlignments updateContigMappingsWithGapSplit(final String contigName, final byte[] contigSeq,
-                                                                                          final GoodAndBadMappings configuration,
-                                                                                          final boolean setResultContigAsAmbiguous) {
-        final GoodAndBadMappings goodAndBadMappings;
-        if ( configuration.goodMappings.stream().anyMatch(alignment -> alignment.containsGapOfEqualOrLargerSize(GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY)) ) {
-            goodAndBadMappings = splitGaps(configuration);
-        } else {
-            goodAndBadMappings = configuration;
-        }
+    private static AssemblyContigWithFineTunedAlignments makeContig(final String contigName, final byte[] contigSeq,
```

Comment:
The name `makeContig` doesn't really capture what this method is doing (we already had contigs coming into this).. maybe something like `createFinalizedConfiguration`?